### PR TITLE
Fixed heat exchanger sizing when used with ZoneHVAC:EnergyRecoveryVentilator

### PR DESCRIPTION
**Pull request by Nigusse**
Addresses #5773 
## Pull request overview

Fixed incorrect sizing of heat recovery heat exchanger generic Design supply air flow rate used in ZoneHVAC:EnergyRecoveryVentilator object and removed duplicate reporting of the heat exchanger design supply air flow rates. Now it is reported from the parent object only. Diff are expected in the sizing results if the design supply air flow rate field of the heat exchanger generic object used in the ZoneHVAC:ERV object is autosized. 
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
  - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [x] CI status: all green or justified
- [x] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [x] Argument types
  - [x] If any virtual classes, ensure virtual destructor included, other things
- [x] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by Nigusse** [comment]
I will add unit test.

**Comment by Nigusse** [comment]
Defect file EIO output before fix:
! <Component Sizing Information>, Component Type, Component Name, Input Field Description, Value
 Component Sizing Information, ZoneHVAC:EnergyRecoveryVentilator, STAND ALONE ERV 1, Design Size Supply Air Flow Rate [m3/s], 0.23600
 Component Sizing Information, Fan:OnOff, STAND ALONE ERV SUPPLY FAN, Design Size Maximum Flow Rate [m3/s], 0.28320
 Component Sizing Information, Fan:OnOff, STAND ALONE ERV EXHAUST FAN, Design Size Maximum Flow Rate [m3/s], 0.28320
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 1, Design Size Nominal Supply Air Flow Rate [m3/s], 0.23600
 Component Sizing Information, ZoneHVAC:EnergyRecoveryVentilator, STAND ALONE ERV 1, Design Size Exhaust Air Flow Rate [m3/s], 0.23600
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 1, Design Size Nominal Supply Air Flow Rate [m3/s], 0.47892
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 1, User-Specified Nominal Supply Air Flow Rate [m3/s], 0.23600

Defect file EIO output after fix:
! <Component Sizing Information>, Component Type, Component Name, Input Field Description, Value
 Component Sizing Information, ZoneHVAC:EnergyRecoveryVentilator, STAND ALONE ERV 1, Design Size Supply Air Flow Rate [m3/s], 0.23600
 Component Sizing Information, Fan:OnOff, STAND ALONE ERV SUPPLY FAN, Design Size Maximum Flow Rate [m3/s], 0.28320
 Component Sizing Information, Fan:OnOff, STAND ALONE ERV EXHAUST FAN, Design Size Maximum Flow Rate [m3/s], 0.28320
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 1, Design Size Nominal Supply Air Flow Rate [m3/s], 0.23600
 Component Sizing Information, ZoneHVAC:EnergyRecoveryVentilator, STAND ALONE ERV 1, Design Size Exhaust Air Flow Rate [m3/s], 0.23600

**Comment by Nigusse** [comment]
@mjwitte Now the sizing is reported from the parent object. See now the eio outputs before and after the fix. Also added unit test. This bug fix is ready for review pending CI machine green outcome.

File: tst/EnergyPlus/unit/HeatRecovery.unit.cc [hunk 1]
```diff
@@ -3953,3 +3953,58 @@ TEST_F( EnergyPlusFixture, SizeHeatRecovery ) {
...
+	ZoneEqSizing( CurZoneEqNum ).DesignSizeFromParent = true;
+	ZoneEqSizing( CurZoneEqNum ).AirVolFlow = 1.0;
+
+	// size the HX nominal supply air volume flow rate 
+	SizeHeatRecovery( ExchNum );
+
+	// verify the name and autosized supply air flow rate
+	EXPECT_EQ( ExchCond( ExchNum ).Name, "HEATRECOVERY HX IN ERV" );
+	EXPECT_EQ( ExchCond( ExchNum ).NomSupAirVolFlow, 1.0 );
+}
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/HeatRecovery.unit.cc]: Need a return at the end of this file.

File: tst/EnergyPlus/unit/HeatRecovery.unit.cc [hunk 2]
```diff
@@ -3953,3 +3953,58 @@ TEST_F( EnergyPlusFixture, SizeHeatRecovery ) {
 	// test autosized face velocity
 	EXPECT_EQ( FaceVelocity, BalDesDehumPerfData( BalDesDehumPerfDataIndex ).NomProcAirFaceVel ); // m/s
 }
+
+TEST_F( EnergyPlusFixture, HeatRecovery_AirFlowSizing ) {
+
+	int ExchNum = 1;
+	int CompanionCoilNum = 0;
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/HeatRecovery.unit.cc]: There's a build warning for this unused variable.

File: tst/EnergyPlus/unit/HeatRecovery.unit.cc (hunk 1 above)
Comment (Nigusse) [inline tst/EnergyPlus/unit/HeatRecovery.unit.cc]: @mjwitte I am not following you what return I need here? Do we always need a return at the end?

File: tst/EnergyPlus/unit/HeatRecovery.unit.cc (hunk 2 above)
Comment (Nigusse) [inline tst/EnergyPlus/unit/HeatRecovery.unit.cc]: Removed the unused variable.

**Comment by Nigusse** [comment]
@mjwitte Addressed your comments and pushed the branch. The CI machine is expected to come clean.

**Comment by mjwitte** [comment]
@Nigusse Tested this with the original 5773 defect file and that is fixed.  Also tested it with one of the defect files from #4338 and found that there are still multiple reports for the HX sizing in the eio output for this file (added to defectfiles\5773):

```
 Component Sizing Information, ZoneHVAC:EnergyRecoveryVentilator, STAND ALONE ERV 1, Design Size Supply Air Flow Rate [m3/s], 7.78779E-002
 Component Sizing Information, Fan:OnOff, STAND ALONE ERV SUPPLY FAN, Design Size Maximum Flow Rate [m3/s], 9.34535E-002
 Component Sizing Information, Fan:OnOff, STAND ALONE ERV EXHAUST FAN, Design Size Maximum Flow Rate [m3/s], 9.34535E-002
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 1, Design Size Nominal Supply Air Flow Rate [m3/s], 7.78779E-002
 Component Sizing Information, ZoneHVAC:EnergyRecoveryVentilator, STAND ALONE ERV 1, Design Size Exhaust Air Flow Rate [m3/s], 7.78779E-002
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 1, Design Size Nominal Supply Air Flow Rate [m3/s], 7.78779E-002
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 1, User-Specified Nominal Supply Air Flow Rate [m3/s], 7.78779E-002

 Component Sizing Information, Fan:OnOff, STAND ALONE ERV SUPPLY FAN 3, Design Size Maximum Flow Rate [m3/s], 1.50000E-002
 Component Sizing Information, Fan:OnOff, STAND ALONE ERV SUPPLY FAN 3, User-Specified Maximum Flow Rate [m3/s], 1.50000E-002
 Component Sizing Information, Fan:OnOff, STAND ALONE ERV EXHAUST FAN 3, Design Size Maximum Flow Rate [m3/s], 1.50000E-002
 Component Sizing Information, Fan:OnOff, STAND ALONE ERV EXHAUST FAN 3, User-Specified Maximum Flow Rate [m3/s], 1.50000E-002
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 3, Design Size Nominal Supply Air Flow Rate [m3/s], 1.50000E-002
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 3, User-Specified Nominal Supply Air Flow Rate [m3/s], 1.50000E-002
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 3, Design Size Nominal Supply Air Flow Rate [m3/s], 1.50000E-002
 Component Sizing Information, HeatExchanger:AirToAir:SensibleAndLatent, OA HEAT RECOVERY 3, User-Specified Nominal Supply Air Flow Rate [m3/s], 1.50000E-002
```

The difference with this test file is that sizing isn't really active (do sizing = no and there aren't any sizing:\* objects) but HX1 and HX2 are autosized and ERV3 and it's fans and HX are hard-sized.  I think the way that the fans are reporting are OK, just the HXs are reporting extra here.

**Comment by Nigusse** [comment]
OK. I will look into it.

**Comment by Nigusse** [comment]
@mjwitte I added logic to catch duplicate reporting of HX sizing and tested the second defect file. It is working fine now.

**Comment by mjwitte** [comment]
All looks good now.  Merging.

**Pull request merged**